### PR TITLE
Update Array's `remove_at()` description to cover removing the last element.

### PR DESCRIPTION
**Pull request by ghost**
Some functions that act on a certain position in an array allow negative positions in order to start from the end. For example, [`rfind()`](https://docs.godotengine.org/en/stable/classes/class_array.html#class-array-method-rfind).

However [`remove_at()`](https://docs.godotengine.org/en/stable/classes/class_array.html#class-array-method-remove-at) does not, so I thought it'd be nice to explicitly explain this in the docs:

> *Note:* `position` cannot be negative. To remove the last element from the array, use `arr.remove_at(arr.size()-1)`.

### CODE REVIEW

**Comment by MewPurPur** [comment]
The new note looks good to me. Do we have any reason not to support negative ones?

**Comment by AThousandShips** [comment]
> Do we have any reason not to support negative positions in this one?

IMO not really, changing this wouldn't break compat as it currently just throws an error, and if someone is depending on this error (somehow, you can't really do it because no error handling is in place) then that's a massively questionable behavior to begin with

TBH I think the fact that the support for negative indices in `Array` is so inconsistent is an issue, and we should really do either or, to prevent confusion and errors when one does work and one does not

**Comment by AThousandShips** [comment]
I'd suggest not just limiting to this but saying something like "to remove an element a certain distance from the end" or something (obviously more clearly worded)

Also I don't think we should suggest this as the appropriate way to remove the last element is `arr.resize(arr.size() - 1)`

File: doc/classes/Array.xml [hunk 1]
```diff
@@ -536,6 +536,7 @@
 				Removes an element from the array by index. If the index does not exist in the array, nothing happens. To remove an element by searching for its value, use [method erase] instead.
 				[b]Note:[/b] This method acts in-place and doesn't return a value.
 				[b]Note:[/b] On large arrays, this method will be slower if the removed element is close to the beginning of the array (index 0). This is because all elements placed after the removed element have to be reindexed.
+				[b]Note:[/b] [param position] cannot be negative. To remove the last element from the array, use [code]arr.remove_at(arr.size()-1)[/code].
```
Comment (AThousandShips) [inline doc/classes/Array.xml:539]:
```suggestion
				[b]Note:[/b] [param position] cannot be negative. To remove an element relative to the end of the array, use [code]arr.remove_at(arr.size()-(i + 1))[/code]. To remove the last element from the array, use [code]arr.resize(arr.size()-1)[/code].
```

Comment (BWPanda) [reply doc/classes/Array.xml]: Thanks. I manually applied this suggestion so I could also fix the typo: `returning the *vale*`.

**Comment by akien-mga** [comment]
Looks good! Could you squash the commits? See [PR workflow](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase) for instructions.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**